File: src/main/java/dominio/metodologias/Antiguedad.java
package dominio.metodologias;

import dominio.empresas.Empresa;
import excepciones.AntiguedadMenorACeroError;

import javax.persistence.Entity;
import java.time.Year;

@Entity
public class Antiguedad extends Cuantificador {

	public Antiguedad() {
	} // PROBLEMA: SE VAN A ESTAR CREANDO MÚLTIPLES INSTANCIAS DE ANTIGUEDAD, CUANDO UNA SERÍA SUFICIENTE (!!!)

	@Override
	public int evaluarEn(Empresa empresa, Year anio) {
		int antiguedad = anio.getValue() - empresa.anioDeCreacion().getValue();
		if (antiguedad < 0) {
			throw new AntiguedadMenorACeroError("La empresa " + empresa.getNombre() + " no existía en el año " + anio);
		}
		return antiguedad;
	}
}

File: src/main/java/dondeInvierto/dominio/metodologias/Antiguedad.kt
package dondeInvierto.dominio.metodologias

import dondeInvierto.dominio.empresas.Empresa
import excepciones.AntiguedadMenorACeroError

import javax.persistence.Entity
import java.time.Year

@Entity
class Antiguedad : Cuantificador() { // TODO tener una única instancia de Antiguedad en la BD

    override fun evaluarEn(empresa: Empresa, anio: Year): Int {
        val antiguedad = anio.value - empresa.anioDeCreacion().value
        if (antiguedad < 0) {
            throw AntiguedadMenorACeroError("La empresa " + empresa.nombre + " no existía en el año " + anio)
        }
        return antiguedad
    }
}